src/cli/cli.cpp

```diff
@@ -261,16 +261,20 @@ Interpreter::Interpreter(Instance *aInstance)
 #endif // OPENTHREAD_CONFIG_DNS_CLIENT_ENABLE
 }
 
-int Interpreter::Hex2Bin(const char *aHex, uint8_t *aBin, uint16_t aBinLength)
+int Interpreter::Hex2Bin(const char *aHex, uint8_t *aBin, uint16_t aBinLength, bool aAllowTruncate)
```

Comment:
Is it necessary to add `aAllowTruncate`? I think we can always allow this. The caller can tell if it's truncated by from the value.